pkg/planner/core/hint_utils.go

```diff
@@ -220,59 +312,185 @@ func getTableName(tblName model.CIStr, asName *model.CIStr) model.CIStr {
 	return tblName
 }
 
-func genJoinMethodHintForSinglePhysicalJoin(sctx base.PlanContext, joinType string, parentOffset int, nodeType h.NodeType, children ...base.PhysicalPlan) (res *ast.TableOptimizerHint) {
+// genJoinMethodHintForSinglePhysicalJoin is the entry point of generating join method hint.
+// It generates a join method hint for a single physical join operator according to the input joinType.
+// Both children of the Join should be passed in as the children arguments, this is for correctly deriving the QB offset
+// for the hint.
+// For hints like merge_join(), we can generate hint using table name of any one of the two tables. But for hints like
+// hash_join_build() and inl_join(), we want to generate hint using table name of a specific side. For this difference,
+// we introduce the onlyFirstTbl argument. If onlyFirstTbl is true, we only try to generate hint using the table name of
+// the children[0].
+func genJoinMethodHintForSinglePhysicalJoin(
+	sctx base.PlanContext,
+	joinType string,
+	parentOffset int,
```

Comment:
Could we rename `parentOffset` to `parentQBOffset` or `parentQueryBlockOffset`? `Offset` seems unclear, better to add keyword `QB` or `QueryBlock` into it.

Reply:
Updated.